### DISCUSSION TITLE
请问着色器可以像VS脚本一样设置为条件启动吗？

### DISCUSSION THREAD

**Discussion by kikka1991** [hooke007/mpv_PlayKit#346] [回收]
我参照作者在vpy的设计与优化思路里提到的方法组合了SVP+SR_ESRGAN_DML的脚本，在720P动画中效果极好，线条清晰瑕疵少；但是应用到1080P动画上会丢失大量细节，AB对比后还是认为Anime4K系列的着色器组合更适合原生1080P动画。

如果要让Anime4K着色器设置自动在1080P源启用，应该如何设置？

import k7sfunc as k7f

clip = video_in

if clip.height <= 720 :
	clip = k7f.FMT_CTRL(clip, h_max=720, fmt_pix=2)
	clip = k7f.ESRGAN_DML(clip, lt_hd=False, model=5005, scale=2, gpu=0, gpu_t=2)
	clip = k7f.FMT_CTRL(clip, h_max=2160)

if container_fps <= 60 :
	clip = k7f.FMT_CTRL(clip, h_max=2160, fmt_pix=2)
	clip = k7f.SVP_PRO(clip, fps_in=container_fps, fps_num=2, fps_den=1, abs=False, cpu=0, nvof=False, gpu=0)

if not (clip == video_in) :
	clip.set_output()

**Comment by hooke007** [comment]
https://hooke007.github.io/unofficial/mpv_profiles.html

**Reply by hooke007** [reply]
esrgan做1080p的放大，不做预压缩720p就不会丢细节。
示例做预压缩是为了省性能。

**Reply by kikka1991** [reply]
感谢回复，参照说明文档折腾了几小时应该是解决问题了。
建议将脚本和着色器的自动启用和条件启用方法例句放在快速说明里，虽然确实是很简单的语句，对我这种外行麻瓜还是相当劝退的，几个条件语句在不同文档里，试错多次才能完全弄懂。

[A4K_Auto]                        
profile-desc=1080P启动A4K着色器    
profile-cond=p["height"]>=1080  
profile-restore=copy              
glsl-shaders-add = "~~/shaders/Anime4K_Clamp_Highlights.glsl;~~/shaders/Anime4K_Restore_CNN_M.glsl;~~/shaders/Anime4K_Upscale_GAN_x2_M.glsl;~~/shaders/Anime4K_AutoDownscalePre_x2.glsl"

**Reply by hooke007** [reply]
快速说明不是给从头学的人看的。
从头学的应该看首页的文档 《win10/11 mpv播放器通用教程》
